app/commands/markdown/preprocess.rb

```diff
@@ -1,9 +1,17 @@
 class Markdown::Preprocess
   include Mandate
 
-  initialize_with :text
+  def initialize(text, strip_h1: true) # rubocop:disable Naming/VariableNumber
+    @text = text.to_s
+    @strip_h1 = strip_h1 # rubocop:disable Naming/VariableNumber
+  end
 
   def call
-    text.gsub(/^`{3,}(.*?)`{3,}\s*$/m) { "\n#{Regexp.last_match(0)}\n" }
+    preprocessed_text = text.gsub(/^`{3,}(.*?)`{3,}\s*$/m) { "\n#{Regexp.last_match(0)}\n" }
+    preprocessed_text = preprocessed_text.gsub(/^# .+?$/, '') if strip_h1 # rubocop:disable Naming/VariableNumber
```

Comment:
I've gone with a very simple replace for h1s.

Reply:
Few thoughts:
1. Could you split this (and the line above it) into methods so that we can see what they do from their names pls?
2. In terms of the regexp, it looks sane, yep.
3. However, I don't think this approach will work. This will also remove any lines containing code with comments (e.g. `# xyz` is the standard format for comments in Ruby), so I think we should actually post-process the H1 out of the resulting HTML, rather than preprocess it here. Which will also mean we'll have to not lower h1 headings in the bit that does that.

Reply:
Actually, thinking about this more, I think there's an easier approach, which is to limit this to the very first line of the file. You'll also need to strip the whitespace line below it.